flowmanager/models.py

```diff
@@ -266,24 +273,27 @@ def list_pipelines(self):
 
     def check_flow_status(self, flow_id):
```

Comment:
The logic in this function is hard to follow, and as you can see also prone to bugs.

Let's try to rewrite in a more understandable way:
If at least one pipeline is running -> RUNNING

Otherwise, we have 7 combinations of (PENDING, SUCCESS, FAILED), we can mark them as:
(P, S, F, PS, PF, SF, PSF)

The mapping between each of the cases to flow states would be:
   1. all * PENDING -> PENDING    [P]
   2. some * PENDING and some * (FAILED / SUCCESS) -> RUNNING     [PF, PS, PSF]
   3. no PENDING and all * SUCCESS -> SUCCESS        [S]
   4. no PENDING and some * (FAILED / SUCCESS) -> FAILED     [F, FS]

You can see we covered all 7 cases.

So, we could write this as this pseudocode
```
if exists pending:
  if exists success or exists failed:
     -> running
  else:
     -> pending
else:
  if exists failed:
      -> failed
  else:
      -> success
```